src/server/middlewares/redirects.ts

```diff
@@ -33,3 +34,26 @@ export const startPageRedirect: IMiddleware<object> = async (ctx) => {
 
   ctx.redirect(`/se${queryStringMaybe}`)
 }
+
+export const manualRedirects: IMiddleware<State, any> = async (ctx, next) => {
+  const permanentRedirects = await getDatasourceEntries('permanent-redirects')
+  const temporaryRedirects = await getDatasourceEntries('temporary-redirects')
+
+  permanentRedirects.datasource_entries.map((redirect) => {
+    if (redirect.name === ctx.request.originalUrl) {
+      ctx.redirect(redirect.value)
+      ctx.status = 301
+      return
+    }
+  })
+
+  temporaryRedirects.datasource_entries.map((redirect) => {
+    if (redirect.name === ctx.originalUrl) {
+      ctx.redirect(redirect.value)
+      ctx.status = 302
+      return
+    }
+  })
```

Comment:
the returns here will just return inside the `map`, not the outer function. Rather than redirecting inside the map, try using for example `find` to locate if there's any redirects matching. I.e.

```ts
const redirect = temporaryRedirects.datasource_entries.find(redirect => redirect.name === ctx.path)
if (redirect) {
  // Perform redirect
}
```

Also i'd suggest we'll use `path` instead of `originalUrl` because if someone gives a random query string we risk losing it. I think we should append the query to the redirect or something rather?

Reply:
Haha how could I miss that, thanks 🤦‍♂️

Ah, that makes sense to add possible query string, added now 👍